# Statistics for Decision Making

## Graded Assignment

### Submitted By
**Sony Banoth**

### Batch
**CPDA B1**

# Question 1

## One Sample Hypothesis Test

**Problem Statement:**

Test whether the average property price in **Altona** is equal to **$800,000** at a **5% significance level**.

In [26]:
# Question 1 - One Sample T-Test

from scipy.stats import ttest_1samp

# Select properties in Altona
altona = df[df['Suburb'] == 'Altona']['Price']

# Remove missing values
altona = altona.dropna()

# Perform one-sample t-test
t_stat, p_value = ttest_1samp(altona, popmean=800000)

print("Number of observations :", len(altona))
print("Sample Mean :", round(altona.mean(), 2))
print("T-Statistic :", round(t_stat, 4))
print("P-Value :", round(p_value, 6))

Number of observations : 74
Sample Mean : 834830.41
T-Statistic : 1.0277
P-Value : 0.307483


### Conclusion

- Significance Level (α) = 0.05
- P-Value = 0.307483

Since the p-value is greater than 0.05, we fail to reject the null hypothesis.

**Conclusion:** There is insufficient statistical evidence to conclude that the average property price in Altona is different from $800,000 at the 5% significance level.

# Question 2

## Independent Two-Sample T-Test

**Problem Statement:**

For the year **2016**, determine whether there is a significant difference in the prices of properties sold during **summer months** and **winter months**.

- Winter months: **October to March**
- Summer months: **April to September**
- Significance Level (α) = **0.05**

In [27]:
# Question 2 - Summer vs Winter Property Prices (2016)

from scipy.stats import ttest_ind

# Convert Date column to datetime
df['Date'] = pd.to_datetime(df['Date'], dayfirst=True)

# Filter data for 2016
df_2016 = df[df['Date'].dt.year == 2016].copy()

# Extract month
df_2016['Month'] = df_2016['Date'].dt.month

# According to the assignment:
# Winter = October to March
# Summer = April to September

winter = df_2016[df_2016['Month'].isin([10, 11, 12, 1, 2, 3])]['Price'].dropna()
summer = df_2016[df_2016['Month'].isin([4, 5, 6, 7, 8, 9])]['Price'].dropna()

# Independent Two-Sample T-Test
t_stat, p_value = ttest_ind(summer, winter, equal_var=False)

print("Summer Properties :", len(summer))
print("Winter Properties :", len(winter))
print("Summer Mean Price :", round(summer.mean(), 2))
print("Winter Mean Price :", round(winter.mean(), 2))
print("T-Statistic :", round(t_stat, 4))
print("P-Value :", round(p_value, 6))

Summer Properties : 4036
Winter Properties : 2300
Summer Mean Price : 1048054.73
Winter Mean Price : 1116647.59
T-Statistic : -3.9211
P-Value : 9e-05


### Hypotheses

- **H₀ (Null Hypothesis):** There is no significant difference between the average property prices in summer and winter during 2016.
- **H₁ (Alternative Hypothesis):** There is a significant difference between the average property prices in summer and winter during 2016.

### Conclusion

- Significance Level (α) = 0.05
- P-Value = 0.00009

Since the p-value is less than 0.05, we reject the null hypothesis.

**Conclusion:** There is sufficient statistical evidence to conclude that the average property prices in summer and winter during 2016 are significantly different.

# Question 3

## Probability of Properties Without Car Parking in Abbotsford

**Problem Statement:**

Calculate the probability that a property located in **Abbotsford** has **no car parking**.

In [28]:
# Question 3 - Probability of No Car Parking in Abbotsford

# Filter properties in Abbotsford
abbotsford = df[df['Suburb'] == 'Abbotsford']

# Total properties
total_properties = len(abbotsford)

# Properties with no car parking (Car = 0)
no_car = len(abbotsford[abbotsford['Car'] == 0])

# Probability
probability = no_car / total_properties

print("Total Properties in Abbotsford :", total_properties)
print("Properties with No Car Parking :", no_car)
print("Probability :", round(probability, 4))

Total Properties in Abbotsford : 56
Properties with No Car Parking : 15
Probability : 0.2679


### Conclusion

- Total Properties in Abbotsford = 56
- Properties with No Car Parking = 15
- Probability = 0.2679

**Conclusion:** The probability that a randomly selected property in Abbotsford has no car parking is **0.2679 (26.79%)**.

# Question 4

## Probability of Selecting a Property with 3 Rooms

**Problem Statement:**

Calculate the probability that a randomly selected property has **3 rooms**.

In [29]:
# Question 4 - Probability of 3-Room Property

# Total number of properties
total_properties = len(df)

# Number of properties with 3 rooms
three_room = len(df[df['Rooms'] == 3])

# Calculate probability
probability = three_room / total_properties

print("Total Properties :", total_properties)
print("Properties with 3 Rooms :", three_room)
print("Probability :", round(probability, 4))

Total Properties : 13580
Properties with 3 Rooms : 5881
Probability : 0.4331


### Conclusion

- Total Properties = 13,580
- Properties with 3 Rooms = 5,881
- Probability = 0.4331

**Conclusion:** The probability that a randomly selected property has **3 rooms** is **0.4331 (43.31%)**.

# Question 5

## Probability of Selecting a Property with 2 Bathrooms

**Problem Statement:**

Calculate the probability that a randomly selected property has **2 bathrooms**.

In [30]:
# Question 5 - Probability of 2 Bathroom Property

# Total number of properties
total_properties = len(df)

# Number of properties with 2 bathrooms
two_bathrooms = len(df[df['Bathroom'] == 2])

# Calculate probability
probability = two_bathrooms / total_properties

print("Total Properties :", total_properties)
print("Properties with 2 Bathrooms :", two_bathrooms)
print("Probability :", round(probability, 4))

Total Properties : 13580
Properties with 2 Bathrooms : 4974
Probability : 0.3663


### Conclusion

- Total Properties = 13,580
- Properties with 2 Bathrooms = 4,974
- Probability = 0.3663

**Conclusion:** The probability that a randomly selected property has **2 bathrooms** is **0.3663 (36.63%)**.

# Question 6

## One Sample T-Test

**Problem Statement:**

Test whether the average property price in **Richmond** is equal to **$1,000,000** at a **5% significance level**.

### Hypotheses

- **H₀ (Null Hypothesis):** The average property price in Richmond is equal to $1,000,000.
- **H₁ (Alternative Hypothesis):** The average property price in Richmond is not equal to $1,000,000.

In [31]:
# Question 6 - One Sample T-Test (Richmond)

from scipy.stats import ttest_1samp

# Select Richmond properties
richmond = df[df['Suburb'] == 'Richmond']['Price'].dropna()

# Perform one-sample t-test
t_stat, p_value = ttest_1samp(richmond, popmean=1000000)

print("Number of Observations :", len(richmond))
print("Sample Mean :", round(richmond.mean(), 2))
print("T-Statistic :", round(t_stat, 4))
print("P-Value :", round(p_value, 6))

Number of Observations : 260
Sample Mean : 1083564.42
T-Statistic : 2.5795
P-Value : 0.010445


### Conclusion

- Significance Level (α) = 0.05
- P-Value = 0.010445

Since the p-value is less than 0.05, we reject the null hypothesis.

**Conclusion:** There is sufficient statistical evidence to conclude that the average property price in Richmond is significantly different from $1,000,000 at the 5% significance level.

# Question 7

## Independent Two-Sample T-Test (Feature Impact)

### Problem Statement

Do properties with car parking sell at a higher average price than properties without car parking across the entire dataset?

Use a **5% significance level** and justify:

- Choice of test
- Interpretation of p-value
- Business implications for developers

### Hypotheses

**H₀ (Null Hypothesis):**
The average selling price of properties with car parking is equal to the average selling price of properties without car parking.

**H₁ (Alternative Hypothesis):**
The average selling price of properties with car parking is different from the average selling price of properties without car parking.

In [32]:
# Question 7 - Independent Two-Sample T-Test (Feature Impact)

from scipy.stats import ttest_ind

# Properties with car parking
with_car = df[df['Car'] > 0]['Price'].dropna()

# Properties without car parking
without_car = df[df['Car'] == 0]['Price'].dropna()

# Perform Welch's Independent T-Test
t_stat, p_value = ttest_ind(with_car, without_car, equal_var=False)

print("Properties with Car Parking :", len(with_car))
print("Properties without Car Parking :", len(without_car))
print("Mean Price (With Car Parking) :", round(with_car.mean(), 2))
print("Mean Price (Without Car Parking) :", round(without_car.mean(), 2))
print("T-Statistic :", round(t_stat, 4))
print("P-Value :", round(p_value, 6))

Properties with Car Parking : 12492
Properties without Car Parking : 1026
Mean Price (With Car Parking) : 1074443.92
Mean Price (Without Car Parking) : 1079088.01
T-Statistic : -0.2722
P-Value : 0.785475


### Choice of Test

An Independent Two-Sample T-Test is used because we are comparing the average selling prices of two independent groups:

- Properties with car parking
- Properties without car parking

### Interpretation of P-Value

- Significance Level (α) = 0.05
- P-Value = 0.785475

Since the p-value is greater than 0.05, we fail to reject the null hypothesis.

### Business Implications for Developers

The statistical analysis indicates that the presence of car parking alone does not significantly increase the average selling price of properties. Developers should focus on other important factors such as location, property type, number of rooms, and overall amenities when planning residential projects.

# Question 8

## Two-Way ANOVA (Location & Property Type)

### Problem Statement

Investigate whether property prices are influenced by:

- Suburb
- Type of Property
- Interaction between Suburb and Property Type

Use a Two-Way ANOVA and explain which factors significantly affect property prices.

### Hypotheses

#### Factor 1: Suburb

**H₀:** The suburb has no significant effect on property prices.

**H₁:** The suburb has a significant effect on property prices.

#### Factor 2: Property Type

**H₀:** Property type has no significant effect on property prices.

**H₁:** Property type has a significant effect on property prices.

#### Interaction Effect

**H₀:** There is no interaction between suburb and property type.

**H₁:** There is a significant interaction between suburb and property type.

In [33]:
# Question 8 - Two-Way ANOVA (Location & Property Type)

from statsmodels.formula.api import ols
import statsmodels.api as sm

# Select required columns
anova_data = df[['Price', 'Suburb', 'Type']].dropna()

# Build Two-Way ANOVA Model
model = ols('Price ~ C(Suburb) + C(Type) + C(Suburb):C(Type)', data=anova_data).fit()

# ANOVA Table
anova_table = sm.stats.anova_lm(model, typ=2)

print(anova_table)

c:\Users\hp\AppData\Local\Python\pythoncore-3.14-64\Lib\site-packages\statsmodels\base\model.py:1894: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 313, but rank is 150
  warnings.warn('covariance of constraints does not have full '
c:\Users\hp\AppData\Local\Python\pythoncore-3.14-64\Lib\site-packages\statsmodels\base\model.py:1894: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 2, but rank is 0
  warnings.warn('covariance of constraints does not have full '
c:\Users\hp\AppData\Local\Python\pythoncore-3.14-64\Lib\site-packages\statsmodels\base\model.py:1923: RuntimeWarning: invalid value encountered in divide
  F /= J


                         sum_sq       df          F         PR(>F)
C(Suburb)          3.776588e+15    313.0  75.183651   0.000000e+00
C(Type)                     NaN      2.0        NaN            NaN
C(Suburb):C(Type)  6.470034e+14    626.0   6.440216  2.505093e-263
Residual           2.068639e+15  12890.0        NaN            NaN


c:\Users\hp\AppData\Local\Python\pythoncore-3.14-64\Lib\site-packages\statsmodels\base\model.py:1894: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 626, but rank is 375
  warnings.warn('covariance of constraints does not have full '


### Conclusion

**Significance Level (α) = 0.05**

#### Effect of Suburb

- P-Value = 0.000000

Since the p-value is less than 0.05, we reject the null hypothesis.

**Conclusion:** The suburb has a statistically significant effect on property prices.

---

#### Effect of Property Type

The ANOVA table could not calculate a separate p-value for **Property Type** because of dependency between variables in the fitted model (shown as NaN). However, property type is included in the interaction model and should be interpreted together with the interaction effect.

---

#### Interaction Effect (Suburb × Property Type)

- P-Value = 2.505093e-263

Since the p-value is less than 0.05, we reject the null hypothesis.

**Conclusion:** There is a statistically significant interaction between suburb and property type, indicating that the effect of property type on price depends on the suburb.

---

### Final Business Conclusion

The Two-Way ANOVA indicates that **location (suburb)** has a significant influence on property prices. In addition, the interaction between **suburb and property type** is also statistically significant. Therefore, developers and real estate businesses should consider both the location and the type of property together when pricing, marketing, or investing in residential properties.

# Question 9

## p-Value Interpretation (Decision Making)

### Problem Statement

A hypothesis test comparing prices across two suburbs results in a **p-value of 0.032**.

Answer the following:

- What does this p-value indicate?
- Should the null hypothesis be rejected at α = 0.05?
- How should a business stakeholder interpret this result?

In [34]:
# Question 9 - P-Value Interpretation

alpha = 0.05
p_value = 0.032

print("Significance Level (α):", alpha)
print("P-Value:", p_value)

if p_value < alpha:
    print("\nDecision: Reject the Null Hypothesis (H₀)")
else:
    print("\nDecision: Fail to Reject the Null Hypothesis (H₀)")

Significance Level (α): 0.05
P-Value: 0.032

Decision: Reject the Null Hypothesis (H₀)


### Interpretation of the p-value

The p-value is **0.032**, which is less than the significance level of **0.05**.

Therefore, we reject the null hypothesis.

### Decision

Since **0.032 < 0.05**, there is sufficient statistical evidence to conclude that the average property prices of the two suburbs are significantly different.

### Business Interpretation

A business stakeholder can conclude that the difference in average property prices between the two suburbs is statistically significant. Therefore, pricing strategies, investment decisions, marketing plans, and property valuations should be developed separately for each suburb rather than assuming that both suburbs have similar market values.

# Question 10

## Industry-Style Hypothesis Validation (Policy Decision)

### Problem Statement

A housing policy group believes that properties with **more than 2 bathrooms** command a premium price.

Design and execute a statistical test to validate this claim.

Clearly state:

- Identify the correct test
- State hypotheses
- Report p-value
- Give a clear recommendation to policymakers

In [35]:
# Question 10 - Industry-Style Hypothesis Validation

from scipy.stats import ttest_ind

# Properties with more than 2 bathrooms
more_than_2 = df[df['Bathroom'] > 2]['Price'].dropna()

# Properties with 2 or fewer bathrooms
two_or_less = df[df['Bathroom'] <= 2]['Price'].dropna()

# Independent Two-Sample T-Test (Welch's Test)
t_stat, p_value = ttest_ind(more_than_2, two_or_less, equal_var=False)

print("Properties with >2 Bathrooms :", len(more_than_2))
print("Properties with <=2 Bathrooms :", len(two_or_less))
print("Mean Price (>2 Bathrooms) :", round(more_than_2.mean(), 2))
print("Mean Price (<=2 Bathrooms) :", round(two_or_less.mean(), 2))
print("T-Statistic :", round(t_stat, 4))
print("P-Value :", round(p_value, 6))

Properties with >2 Bathrooms : 1060
Properties with <=2 Bathrooms : 12520
Mean Price (>2 Bathrooms) : 1882824.2
Mean Price (<=2 Bathrooms) : 1007347.94
T-Statistic : 28.8002
P-Value : 0.0


### Correct Statistical Test

An **Independent Two-Sample T-Test (Welch's T-Test)** is used because we are comparing the average selling prices of two independent groups:

- Properties with more than 2 bathrooms
- Properties with 2 or fewer bathrooms

### Hypotheses

**Null Hypothesis (H₀):**
Properties with more than 2 bathrooms do **not** have a significantly higher average selling price than properties with 2 or fewer bathrooms.

**Alternative Hypothesis (H₁):**
Properties with more than 2 bathrooms have a significantly higher average selling price than properties with 2 or fewer bathrooms.

### Test Result

- Significance Level (α) = 0.05
- P-Value = 0.000000

Since the p-value is less than 0.05, we reject the null hypothesis.

### Recommendation to Policymakers

The statistical analysis provides strong evidence that properties with more than two bathrooms command significantly higher average selling prices. Therefore, policymakers and housing planners should consider the number of bathrooms as an important factor when developing housing policies, planning residential projects, and estimating property values.